lib/mongodb-filter-wrapper.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+class MongoDBFilterWrapper {
+	static get filtersAllowed() {
+		return {
+			equal: '$eq',
+			not: '$ne',
+			greater: '$gt',
+			greaterOrEqual: '$gte',
+			lesser: '$lt',
+			lesserOrEqual: '$lte',
+			in: '$in',
+			notIn: '$nin',
+			reg: '$regex',
+			all: '$all'
+		};
+	}
+
+	/**
+		* Enter door to filter process
+		* @param model {Object} Model that will use the filter
+		* @param filters {Object} Filters defined as parameters
+		* @returns {Object} with the filter parsed
+		*/
+	static process(model, filters) {
+
+		// Validate filters given
+		if(typeof filters !== 'object' || (Array.isArray(filters) && filters.length === 0) || Object.entries(filters).length === 0)
+			return {};
+
+		// Get model filters
+		const modelFields = model.constructor.fields || {};
+
+		return this.getParsedFilters(filters, modelFields);
+	}
+
+	/**
+		* Validates filters given and loop calling to parse
+		* @param filters {Object} Filters defined as parameters
+		* @param modelFields {Object} Model that will use the filter
+		* @returns {Array|Object} The filter parsed to use in MongoDB
+		*/
+	static getParsedFilters(filters, modelFields) {
+		const orCondition = Array.isArray(filters);
+
+		let parsedFilters = orCondition ? [] : {};
+
+		// Only arrays as a filter
+		if(filters !== undefined && typeof filters === 'object' && Object.keys(filters).length > 0 && !Array.isArray(filters))
+			filters = [filters];
+
+		if(filters && filters.length > 0) {
+			// First loop over all filters given by parameters
+			filters.forEach(filterGiven => {
+				const fullyParsedFilter = this.parseObjectFilter(filterGiven, modelFields);
+				parsedFilters = (orCondition && Array.isArray(parsedFilters))
+					? [...parsedFilters, fullyParsedFilter] : Object.assign(parsedFilters, fullyParsedFilter);
+			});
+		}
+
+		if(orCondition && parsedFilters.length > 0) {
```

Comment:
```suggestion
		if(orCondition && parsedFilters.length) {
```
:policeman: no es necesario validar que sea mayor que cero